packages/frontend/src/types/Events.ts

```diff
@@ -0,0 +1,36 @@
+export enum PageType {
+    PAGE = 'page',
+    MODAL = 'modal',
+}
+
+export enum PageName {
+    REGISTER = 'register',
+    LOGIN = 'login',
+    SIGNUP = 'signup',
+    EXPLORER = 'explorer',
+    EXPLORE_TABLES = 'explore_tables',
+    SAVED_QUERIES = 'saved_charts',
+    SAVED_QUERY_EXPLORER = 'saved_chart_explorer',
+    PROFILE_SETTINGS = 'profile_settings',
+    PASSWORD_SETTINGS = 'password_settings',
+    ORGANIZATION_SETTINGS = 'organization_settings',
+    USER_MANAGEMENT_SETTINGS = 'user_management_settings',
+    ABOUT_LIGHTDASH = 'about_lightdash',
+}
+
+export enum CategoryName {
+    SETTINGS = 'settings',
+}
+
+export enum SectionName {
+    EMPTY_RESULTS_TABLE = 'empty_results_table',
+    EXPLORER_TOP_BUTTONS = 'explorer_top_buttons',
+    SIDEBAR = 'sidebar',
+}
+
+export enum EventName {
```

Comment:
With this approach can we still strongly type the allowed `properties` for a given event type?

Reply:
Having an enum for the event names won't interfere with that. 
If we want stronger typing for event properties we would need to change the `EventData` type to:
```

type GenericEventData = {
    name: EventName;
    properties?: Record<string, any>;
}

type SpecificEventData = {
    name: EventName.MyEventName;
    properties: {
         myUuid: string;
    };
}

type EventData = GenericEventData | SpecificEventData;

```
But currently there is no FE event with properties.